src/core/geometry/qgsgeometry.cpp

```diff
@@ -3174,6 +3174,34 @@ QgsGeometry QgsGeometry::forceRHR() const
   return forcePolygonClockwise();
 }
 
+Qgis::AngularDirection QgsGeometry::polygonOrientation() const
+{
+  if ( !d->geometry )
+  {
+    return Qgis::AngularDirection::Clockwise;
+  }
+
+  if ( isMultipart() )
+  {
+    const QgsGeometryCollection *collection = qgsgeometry_cast< const QgsGeometryCollection * >( d->geometry.get() );
+    const QgsAbstractGeometry *g = collection->geometryN( 0 );
```

Comment:
is it safe to call geometryN(0) on an empty geometry ?

Reply:
Yes, it will return NoOrientation here